locallib.php

```diff
@@ -450,7 +450,7 @@ function auth_saml2_get_idps($active = false, $asarray = false) {
  * @return object The default IdP object, or NULL if there is no default IdP set.
```

Comment:
this phpdoc needs a slight tweak now

Can you please wrap this in unit tests. I'm not sure this is correct, it appears in some cases to return an object, and in other cases to return an entityid

Reply:
Hi @brendanheywood , what would you suggest for the phpdoc? It's still possible for that function to return null if we have no idp's set at all.

Reply:
I have added a unit test, and fixed that function and the place in config/authsources.php where we need to get the entity id from the returned object. https://github.com/catalyst/moodle-auth_saml2/pull/344/files#diff-36bb225ac3d28fa53bb66e657ddefdcaR58